featuremanagement/_featuremanager.py

```diff
@@ -100,9 +103,11 @@ def _check_default_disabled_variant(feature_flag):
         """
         if not feature_flag.allocation:
             return EvaluationEvent(enabled=False)
-        return FeatureManager._check_variant_override(
+        evaluation_event = FeatureManager._check_variant_override(
             feature_flag.variants, feature_flag.allocation.default_when_disabled, False
         )
+        evaluation_event.reason = VariantAssignmentReason.DEFAULT_WHEN_DISABLED
```

Comment:
When feature_flag.allocation is None and evaluation_event.enabled == false, the assigned variant should be `None` and reason should be "DefaultWhenDisabled". @jimmyca15 [ref](https://github.com/microsoft/FeatureManagement-Dotnet/pull/290#discussion_r1430793880)

Currently, if not feature_flag.allocation, this code path will be skipped at line 104 and the assignment reason will be set to "None".


Please refer to this [comment](https://github.com/microsoft/FeatureManagement-Python/pull/24#issuecomment-2106960349) and this [PR](https://github.com/microsoft/FeatureManagement-Dotnet/pull/290) for details.

Reply:
This should be fixed now.